pkg/services/notification/controller.go

```diff
@@ -23,33 +26,61 @@ import (
 type Controller struct {
 	runningTaskIds         chan string
 	runningNotificationIds chan string
-	taskQueue              lib.Topic
-	notificationQueue      lib.Topic
+	taskQueue              q.IQueue
+	notificationQueue      q.IQueue
+	websocketMsgChan       chan string
 }
 
 func NewController() (*Controller, error) {
-	cfg := config.GetInstance()
-	queueConnStr := cfg.Queue.Addr
-	queueType := cfg.Queue.Type
-
-	queueConfigMap := map[string]interface{}{
-		"connStr": queueConnStr,
-	}
-
-	c, err := q.New(queueType, queueConfigMap)
-	if err != nil {
-		logger.Errorf(nil, "Failed to connect redis queue: %+v.", err)
-		return nil, err
+	iClient := global.GetInstance().GetQueueClient()
+
+	queueType := config.GetInstance().Queue.Type
+	var iqueue q.IQueue
+	var ipubsub q.IPubSub
+	var notificationQueue q.IQueue
+	var taskQueue q.IQueue
+	var wsMsgStrChan chan string
+
+	if queueType == constants.QueueTypeRedis {
+		//queue setting
+		redisQueue := qredis.RedisQueue{}
+		iqueue = &redisQueue
+
+		//pubsub setting
+		if config.GetInstance().Websocket.Enable {
+			redisPubSub := qredis.RedisPubSub{}
```

Comment:
Should not care about queueType: redis or etcd. 

ipubsub = queue.NewPubSub(queueType) will be better.

Reply:
fix